[FEAT] ID 찾기 / PW 변경 API 추가

jye326 opened the pull request:
## Issue Number
closed #1159 

- ID 찾기 및 PW 변경 API를 추가합니다.

## To-Be
<!-- 변경 사항 -->
1. Phone 객체와 전화번호(String)를 구분하기 위해 RequestDTO 필드명 수정 `String phone` -> `String phoneNumber`
2. 전체 코드 포맷팅(우테코 스타일 XML 자동 정렬)
3. 회원 가입 시 전화번호 인증 여부 검증 로직 추가
4. 이름과 전화번호로 로그인 아이디를 조회하는 API 추가
5. 기존 회원가입 서비스 메서드`AuthService.register()`가 PresentationLayerDto를 그대로 사용하고 있어 계층 분리 및 리팩터링
6. Password 객체와 필드(암호화된 실제 문자열 값)를 구분하기 위해 필드명 수정 및 `@Column`명 지정 `Password.password` -> `Password.value`
7. 인증 확인 된 전화 번호와 로그인 아이디로 비밀번호를 변경하는 API 추가

## 추신
- 리팩터링 등으로 변경 내용이 많습니다. Commit 별로 확인하며 리뷰하시는 것을 추천합니다.

## Check List
- [x] 테스트가 전부 통과되었나요?
- [x] 모든 commit이 push 되었나요?
- [x] merge할 branch를 확인했나요?
- [x] Assignee를 지정했나요?
- [x] Label을 지정했나요?
- [x] 닫을 이슈 번호를 지정했나요?

jye326 committed:
feat: 전화번호 인증 로직 및 회원가입 테스트 개선

- `phone` 필드를 `phoneNumber`로 변경하여 이름 명확성 개선
- 전화번호 인증 여부 확인 로직 추가
- 기존 테스트 코드에서 전화번호 인증 시나리오 추가 및 수정
- `BusinessErrorMessage`에서 인증 실패 메시지 수정
- `PhoneVerificationService`에 유효한 인증 정보 존재 여부 검증 메서드 추가
- `AuthService`에서 회원가입 시 전화번호 인증 여부 확인 로직 연동
- 관련 API 문서 수정 및 추가
- 불필요한 주석 및 코드 정리

jye326 committed:
refactor: 코드 정렬 및 형식 수정

- 누락된 공백 및 들여쓰기 정리
- Test 코드 및 DTO 파일 등 전반적인 포맷팅 수정
- 가독성 향상을 위한 코드 스타일 일관성 유지
- 불필요한 코드 삭제 및 구조 개선

jye326 committed:
feat: 로그인 아이디 찾기 기능 추가 및 관련 API 수정

- 이름과 전화번호로 로그인 아이디를 조회하는 기능 추가
- `/validate-id`를 `/validate-login-id`로 엔드포인트 수정
- `LoginIdResponse` 및 `FindLoginIdRequest` DTO 추가
- `LOGIN_ID_NOT_FOUND` 에러 메시지 수정 및 코드 정리
- 관련 테스트 코드 및 통합 테스트 추가

jye326 committed:
refactor: FindLoginIdRequest 유효성 검증 로직 추가

- name 필드에 @Size 어노테이션 추가로 길이 제한 (2자 이상 5자 이하) 적용
- phoneNumber 필드에 @PhoneNumber 어노테이션 추가로 유효성 검증 강화

jye326 committed:
refactor: AuthController 및 AuthService 리팩토링

- 불필요한 import 정리 및 wildcard(*) 사용으로 가독성 향상
- AuthService의 register 메서드 구조 간소화: SignUpRequest를 개별 값 인자로 분리
- createMember 로직 분리 및 기존 중복된 메서드 제거
- 관련 테스트 코드 수정 및 리팩토링

jye326 committed:
refactor: Member 및 Password 클래스 메서드 명칭 변경 및 관련 테스트 수정

- `Member#getPassword`를 `Member#getPasswordValue`로 변경하여 명확성 향상
- `Password#password` 필드 명칭을 `value`로 변경
- 관련 테스트 코드 수정 및 검증 로직 반영

jye326 committed:
feat: 비밀번호 초기화 API 추가 및 관련 로직 구현

- AuthController에 resetPassword 엔드포인트 추가
- ResetPasswordRequest DTO 생성 및 유효성 검증 로직 반영
- AuthService에 비밀번호 초기화 로직 추가 및 Member 패스워드 업데이트 로직 활용
- 관련 테스트 코드 작성 및 기존 테스트 수정

ppparkta commented on backend/fittoring/src/main/java/fittoring/application/auth/presentation/AuthController.java:
```diff
@@ -59,7 +55,7 @@ public class AuthController {
     @PostMapping("/signup")
     public ResponseEntity<Void> signUp(@RequestBody @Valid SignUpRequest request) {
-        authService.register(request);
+        authService.register(request.loginId(), request.name(), request.gender(), request.phoneNumber(), request.password());
```
p1: DTO를 바로 인자로 전달하는 것이 유지보수 측면에서 좋지 않을까요?

jye326 replied:
1. 애플리케이션 계층의 요청 DTO를 서비스 메서드에 직접 전달하게 되면 파라미터가 깔끔해지는 장점이 있습니다.
또한 DTO에 필드가 추가되어도 메서드 시그니처를 수정하지 않아도 되죠!

2. 하지만 서비스 레이어의 메서드가 애플리케이션 레이어의 DTO와 강결합 되는 문제가 있습니다.
이는 메서드의 재사용성을 낮추고 단위 테스트 작성 시에도 불필요한 DTO를 생성하도록 강제합니다. 계층 간 독립성을 떨어뜨리는 것이죠.
이러한 이유로 이전에 계층 분리를 위해 각 Layer의 DTO별로 네이밍 양식과 패키지 상 위치의 차를 둔 것으로 기억합니다.

- SRP 관점으로도 1번 방식은 서비스 메서드가 비즈니스 로직의 변경이 아닌 UI/API 스펙의 변경을 원인으로 수정되어야 하므로 2번 방식이 더 객체 지향적이라고 볼 수 있습니다.
- 그러나 2번 방식은 메서드 파라미터가 10~20개 되도록 많아지면 가독성이 심각하게 저하되는 문제가 있습니다. 이러한 상황에서 실무에서는 RequestDto를 컨트롤러에서 서비스 계층의 입력값 전용 DTO(주로 Command라는 이름)로 변환하여 넘긴다고 하네요
- 현재처럼 파라미터 5개 이하 수준이면 2번 방식이 적합하다고 생각합니다.

현재 프로덕션 코드는 도메인 영역 전반에 두 가지 방식이 혼재되어 있는 상황입니다만 점진적으로 결합도를 낮추는 리팩터링을 진행하면 좋을 것 같습니다.

ppparkta replied:
p2: 계층 간 결합을 낮추는 의도 좋네요!👍
이번 pr에 컨트롤러 계층에서 서비스레이어의 dto를 만들어서 넘겨주면 좋을 것 같아요 잘 지켜지진 않았지만 플젝 초기에 매개변수 3개 이상일 때 서비스 레이어 dto를 만들자고 이야기했었으니, 이번 기회에 코드에 반영해주시고 다른 코드들도 차차 수정해 나가는거 어떠신가요?

kimdevv replied:
저도 5개는 너무 많고, 2~3개 이상일 경우에 서비스 Dto 만드는 거에 찬성합니다!

dye0p replied:
2번 방식이 계층 간 결합을 낮추는 것에는 좋다고 생각하지만, 현재 `loginId`와 `name`이 모두 `String 타입`으로 연속되어 있어, 호출부에서 인자 순서가 바뀌더라도 컴파일 에러가 발생하지 않는 위험이 있을 것 같아요. 이전에 채팅 부분에서 이러한 경우가 있어서 런타임에 예외가 발생했던 적이 있어요!

이 부분을 조심해서 작성하면 좋겠지만, 그렇지 않을 수도 있으니 VO로 포장하거나, 1개 이상일 땐 DTO로 넘겨주는 것이 좋을 것 같습니다!

jye326 replied:
ServiceDto로 넘기도록 변경하였습니다~

ppparkta commented on backend/fittoring/src/main/java/fittoring/application/auth/presentation/AuthController.java:
```diff
@@ -181,4 +177,17 @@ public ResponseEntity<LoginResponse> oauthSignUp(
+
+    @GetMapping("/find-login-id")
+    public ResponseEntity<LoginIdResponse> findLoginId(@RequestBody @Valid FindLoginIdRequest request) {
+        String loginId = authService.findLoginId(request.name(), request.phoneNumber());
```
p1: 여기도 DTO를 그대로 서비스 레이어에 보내는 것이 좋지 않을까요?

jye326 replied:
위와 같습니다~!

ppparkta commented on backend/fittoring/src/main/java/fittoring/application/auth/presentation/AuthController.java:
```diff
@@ -181,4 +177,17 @@ public ResponseEntity<LoginResponse> oauthSignUp(
+
+    @PostMapping("/reset-password")
+    public ResponseEntity<Void> resetPassword(@RequestBody @Valid ResetPasswordRequest request) {
+        authService.resetPassword(request.loginId(), request.phoneNumber(), request.password());
```
p1: 여기도 DTO를 그대로 서비스 레이어에 보내는 것이 좋지 않을까요?

jye326 replied:
이것도!

ppparkta commented on backend/fittoring/src/main/java/fittoring/application/auth/service/AuthService.java:
```diff
@@ -172,4 +166,26 @@ private Member getRandomIdPwMember(OauthSignUpRequest request) {
+        if (!member.getName().equals(name)) {
+            throw new MemberNotFoundException(LOGIN_ID_NOT_FOUND_MESSAGE);
+        }
+        return member.getLoginId();
```
p3: 주석 없이도 코드가 잘 이해돼서 제거해도 될 것 같아요 👍

ppparkta commented on backend/fittoring/src/main/java/fittoring/application/auth/service/AuthService.java:
```diff
@@ -172,4 +166,26 @@ private Member getRandomIdPwMember(OauthSignUpRequest request) {
+                .orElseThrow(() -> new MemberNotFoundException(BusinessErrorMessage.MEMBER_NOT_FOUND.getMessage()));
+        // 패스워드 변경
+        member.updatePassword(password);
+        return member;
```
p3: 여기도 마찬가지로 이해가 잘 돼서 주석이 없어도 될 것 같아요!

kimdevv replied:
동의합니다.

ppparkta commented on backend/fittoring/src/main/java/fittoring/application/auth/service/PhoneVerificationService.java:
```diff
@@ -38,17 +38,24 @@ private LocalDateTime calculateExpiredTime() {
+        }
+    }
+
+    public void existValidPhoneVerification(Phone phone) {
```
p2: 해당 부분 검증에 대해서 확인 차 질문 남깁니다!

전화번호 인증하기 버튼을 눌렀을 때 PhoneVerification 행을 검사해도 인증 완료가 되었는지는 검증할 수 없는 상태인 것 같아요. 

지금은 인증번호 발급까지만 진행하고 인증하지 않은 상태로 회원가입을 해도 회원가입이 가능한 상태인 것 같은데 확인 부탁드립니다!

kimdevv replied:
그렇네요

jye326 replied:
- `PhoneVerification`(이하 PV)에 `isVarified` : boolean 필드를 추가했습니다. 이는 인증이 되었는지 여부를 판단하는 필드입니다.
- `verifyCode` 에서 유효 기간 내이며, 이전에 인증하지 않은 PV객체를 검증 요청 시 `isVarified` -> ture로 인증 체크를 하도록 하였습니다.
- `existValidPhoneVerification` (현재는 `checkVerificationStatus`로 이름 변경) 에서는 Phone이 일치하는 PV 탐색, 유효 기간 검증, `isVarified` true 를 확인하도록 하여 이전에 `verifyCode` 메서드를 통해 인증한 PV임을 검증합니다.
---

> - `createPhoneVerification` : PV 생성 및 code 생성
> - `verifyCode` : code를 통한 PV 검증

위 두 메서드는 `@Transactional`을 추가하였습니다.

> - `checkVerificationStatus` : Phone 일치하는 유효한 PV 조회/검증

위 메서드는 조회 작업만 발생하므로 `@Transactional(readonly=true)`처리하였습니다.

ppparkta reviewed (commented):
고생하셨습니다. 리뷰 몇 개 남겨두었습니다! 
확인 후 바로 승인할게요 👍

kimdevv commented on backend/fittoring/src/main/java/fittoring/application/auth/service/PhoneVerificationService.java:
```diff
@@ -17,11 +17,11 @@
 @Service
 public class PhoneVerificationService {
 
+    private static final String INVALID_PHONE_VERIFICATION_MESSAGE = BusinessErrorMessage.PHONE_VERIFICATION_INVALID.getMessage();
```
p1: `BusinessErrorMessage` 값들이 상수 처리된 부분이 많은 것 같아요. 해당 부분들을 상수로 분리해내신 이유가 궁금합니다.

jye326 replied:
클래스 내에서 동일한 에러 메시지를 여러 번 사용하는데 호출자(.)가 두번이라 길더라구요. static import는 컨벤션으로 하지 않기로 했으니 가독성 개선을 위해 상수로 뺐습니다

kimdevv commented on backend/fittoring/src/main/java/fittoring/application/auth/service/AuthService.java:
```diff
@@ -31,12 +30,16 @@ public class AuthService {
     private final MemberOauthRepository memberOAuthRepository;
+    private final PhoneVerificationService phoneVerificationService;
+
+    private static final String LOGIN_ID_NOT_FOUND_MESSAGE = BusinessErrorMessage.LOGIN_ID_NOT_FOUND.getMessage();
```
p1: 여기도요!

kimdevv reviewed (commented):
고생하셨습니다~ 몇 가지 커멘트 남깁니다!

jye326 committed:
refactor: 회원 가입 로직 리팩토링 및 DTO 도입

- 회원 가입 요청 처리에 RegisterMemberDto 적용
- AuthService 및 AuthController의 register 메서드 구조 개선
- SignUpRequest에 toRegisterMemberDto 메서드 추가
- 관련 테스트 코드 수정 및 리팩토링

dye0p commented on backend/fittoring/src/main/java/fittoring/application/auth/service/AuthService.java:
```diff
@@ -52,6 +55,16 @@ private void validateDuplicatePhone(String phone) {
+                Password.from(password)
+        );
+    }
+
```
p3: 왜 위로 올리신거죠?

jye326 replied:
register 메서드에서 호출되는 순서에 따라 읽을 수 있도록 재배치하였습니다.

dye0p commented on backend/fittoring/src/main/java/fittoring/application/auth/service/AuthService.java:
```diff
@@ -163,4 +166,26 @@ private Member getRandomIdPwMember(OauthSignUpRequest request) {
                 Password.from(randomPw)
         );
     }
+
+    public String findLoginId(String name, String phoneNumber) {
```
p1: `@Transactional(readOnly = true)` 을 붙여주면 좋을 것 같습니다 :)

jye326 replied:
추가했습니다!

dye0p commented on backend/fittoring/src/main/java/fittoring/application/auth/presentation/AuthController.java:
```diff
@@ -172,4 +177,17 @@ public ResponseEntity<LoginResponse> oauthSignUp(
                 .body(response);
     }
+
+    @GetMapping("/find-login-id")
```
p1: 이미 GET 메서드를 통해서 자원을 가져온다는 의미가 전달되니 `find-`는 제거해도 될 것 같다고 생각합니다. 어떻게 생각하시나요!

jye326 replied:
로그인 id를 단순히 조회하는 것이 아니라 Phone을 통해 찾는 것이기 때문에 의미를 드러내야 할 것 같습니다. 
만약 로그인 한 유저가 본인의 로그인 id를 조회하는 api가 있다고 하면 그때 GET login-id이 적합할 것 같습니다.

dye0p replied:
무엇으로 찾을지는 요청 body를 통해서 알 수 있다고 생각하는데, 어떻게 생각하시나요~

jye326 replied:
반영했습니다~

dye0p commented on backend/fittoring/src/main/java/fittoring/application/auth/presentation/AuthController.java:
```diff
@@ -172,4 +177,17 @@ public ResponseEntity<LoginResponse> oauthSignUp(
+                .body(new LoginIdResponse(loginId));
+    }
+
+    @PostMapping("/reset-password")
```
p1: 이 엔드포인트도 `reset-`이 꼭 필요할까요?

jye326 replied:
개인적으로 POST password는 비밀번호 초기화의 의미가 선명하지 않다고 느껴집니다!

dye0p commented on backend/fittoring/src/main/java/fittoring/application/auth/presentation/AuthController.java:
```diff
@@ -172,4 +177,17 @@ public ResponseEntity<LoginResponse> oauthSignUp(
+    @PostMapping("/reset-password")
+    public ResponseEntity<Void> resetPassword(@RequestBody @Valid ResetPasswordRequest request) {
+        authService.resetPassword(request.loginId(), request.phoneNumber(), request.password());
+        return ResponseEntity.status(HttpStatus.OK).build();
```
p2: 응답 body가 없으니 상태코드 204는 어떨까요?

jye326 replied:
반영했습니다~

dye0p commented on backend/fittoring/src/main/java/fittoring/application/auth/presentation/AuthController.java:
```diff
@@ -172,4 +177,17 @@ public ResponseEntity<LoginResponse> oauthSignUp(
+    @GetMapping("/find-login-id")
+    public ResponseEntity<LoginIdResponse> findLoginId(@RequestBody @Valid FindLoginIdRequest request) {
+        String loginId = authService.findLoginId(request.name(), request.phoneNumber());
+        return ResponseEntity.status(HttpStatus.OK)
+                .body(new LoginIdResponse(loginId));
```
p3: 사소하지만 이렇게도 할 수 있을 것 같아요! 
```suggestion
        return ResponseEntity.ok(new LoginIdResponse(loginId));
```
`ok()`는 내부적으로 `HttpStatus.OK`를 포함하고 있습니다!

<img width="516" height="168" alt="Image" src="https://github.com/user-attachments/assets/4e8fb152-7de2-4e51-a61c-3af8ce408cff" />

이렇게 하면 가독성이 조금! 더 올라갈 것 같네요 : )

jye326 replied:
컨벤션을 통일하기 위해 status를 드러내고 있습니다!

dye0p commented on backend/fittoring/src/main/java/fittoring/application/auth/service/AuthService.java:
```diff
@@ -163,4 +166,26 @@ private Member getRandomIdPwMember(OauthSignUpRequest request) {
+            throw new MemberNotFoundException(LOGIN_ID_NOT_FOUND_MESSAGE);
+        }
+        return member.getLoginId();
+    }
+
+    public Member resetPassword(String loginId, String phoneNumber, String password) {
```
p1: `@Transactional` 을 붙여주면 좋을 것 같습니다 :)

dye0p commented on backend/fittoring/src/main/java/fittoring/application/auth/service/AuthService.java:
```diff
@@ -163,4 +166,26 @@ private Member getRandomIdPwMember(OauthSignUpRequest request) {
+        phoneVerificationService.existValidPhoneVerification(new Phone(phoneNumber));
+        // loginId(unique)로 Member 탐색
+        Member member = memberRepository.findByLoginId(loginId)
+                .orElseThrow(() -> new MemberNotFoundException(BusinessErrorMessage.MEMBER_NOT_FOUND.getMessage()));
```
p2: 105 ~ 107번 라인에서 `getMemberByLoginId()`가 동일한 기능을 수행하고 있어요! 예외 메세지가 다른데, 의도하신 걸까요?

jye326 replied:
예외 클래스명과 예외 메시지가 어순에 맞지 않는 것 같아 새로 만들었습니다;; 변경해서 통일하겠습니다.

dye0p commented on backend/fittoring/src/main/java/fittoring/application/auth/service/AuthService.java:
```diff
@@ -163,4 +166,26 @@ private Member getRandomIdPwMember(OauthSignUpRequest request) {
+                .orElseThrow(() -> new MemberNotFoundException(BusinessErrorMessage.MEMBER_NOT_FOUND.getMessage()));
+        // 패스워드 변경
+        member.updatePassword(password);
+        return member;
```
p1: 리턴된 member를 컨트롤러에서 사용하지 않고 있는데 의도가 궁금합니다!

jye326 replied:
서비스 테스트에서 변경된 비밀번호를 검증하기 위함입니다. 아니면 DB조회를 해서 변경된 값을 확인해야 합니다. 다른 좋은 방법이 있을까요?

dye0p replied:
테스트에서 직접 `findBy~`를 해주는 방법이 있을 것 같아요!

dye0p commented on backend/fittoring/src/main/java/fittoring/application/auth/presentation/AuthController.java:
```diff
@@ -50,7 +55,7 @@ public class AuthController {
     @PostMapping("/signup")
     public ResponseEntity<Void> signUp(@RequestBody @Valid SignUpRequest request) {
-        authService.register(request);
+        authService.register(request.loginId(), request.name(), request.gender(), request.phoneNumber(), request.password());
```
p2: 타입 안정성을 위해서 `phoneNumber`를 포장해서 넘겨주면 어떨까요?

```suggestion
        authService.register(request.loginId(), request.name(), request.gender(), new Phone(request.phoneNumber()), request.password());
```

현재 컨트롤러에서 `phoneNumber`를 원시 타입으로 넘겨주는 모든 부분에도 해당하는 코멘트 입니다!

dye0p commented on backend/fittoring/src/main/java/fittoring/application/auth/service/PhoneVerificationService.java:
```diff
@@ -38,17 +38,24 @@ private LocalDateTime calculateExpiredTime() {
+                .orElseThrow(() -> new InvalidPhoneVerificationException(INVALID_PHONE_VERIFICATION_MESSAGE));
+        LocalDateTime now = LocalDateTime.now(ZoneId.of("Asia/Seoul"));
+        if (phoneVerification.isExpired(now)) {
+            new InvalidPhoneVerificationException(INVALID_PHONE_VERIFICATION_MESSAGE);
```
p1: `throw` 하고 있지 않네요!

jye326 replied:
어머 실수!

dye0p commented on backend/fittoring/src/main/java/fittoring/application/auth/service/PhoneVerificationService.java:
```diff
@@ -38,17 +38,24 @@ private LocalDateTime calculateExpiredTime() {
+                .orElseThrow(() -> new InvalidPhoneVerificationException(INVALID_PHONE_VERIFICATION_MESSAGE));
+        LocalDateTime now = LocalDateTime.now(ZoneId.of("Asia/Seoul"));
+        if (phoneVerification.isExpired(now)) {
+            new InvalidPhoneVerificationException(INVALID_PHONE_VERIFICATION_MESSAGE);
```
p2: 48~50 라인과 동일하네요! 메서드로 추출 하면 좋을 것 같아요!

jye326 replied:
변경되었습니다!

dye0p commented on backend/fittoring/src/main/java/fittoring/application/auth/service/PhoneVerificationService.java:
```diff
@@ -38,17 +38,24 @@ private LocalDateTime calculateExpiredTime() {
+        }
+    }
+
+    public void existValidPhoneVerification(Phone phone) {
```
p1: 호출하는 쪽 트랜잭션에 합류하기 위해서 `@Transactional(readOnly = true)` 을 붙여주면 좋을 것 같습니다 :)

dye0p commented on backend/fittoring/src/main/java/fittoring/application/auth/service/PhoneVerificationService.java:
```diff
@@ -38,17 +38,24 @@ private LocalDateTime calculateExpiredTime() {
     }
 
     public void verifyCode(VerificationCodeRequest request) {
```
p1: 이번 PR의 변경사항은 아니지만 `@Transactional(readOnly = true)` 을 적용해주면 좋을 것 같아요!

dye0p reviewed (changes requested):
리뷰하다보니 코멘트가 꽤 많아졌네요.. 확인해주시면 감사하겠습니다! 수고많으셨어요~

jye326 committed:
feat: 휴대폰 인증 상태 검증 로직 추가 및 관련 코드 수정

- phone_verification 테이블에 is_verified 컬럼 추가
- 인증 상태 확인 및 갱신을 위한 verify 및 checkVerificationStatus 메서드 구현
- AuthService 및 PhoneVerificationService의 작업 로직 수정
- 회원 가입 및 비밀번호 초기화 시 인증 상태 검증 로직 반영

jye326 committed:
refactor: FixtureUtil 개선 및 비밀번호 초기화 로직 리팩토링

- FixtureUtil에 getVerifiedPhoneVerification 메서드 추가
- resetPassword 반환 상태를 HttpStatus.NO_CONTENT로 수정
- resetPassword 메서드에 @Transactional 적용
- 테스트 코드에서 FixtureUtil 사용으로 코드 중복 제거 및 가독성 향상

jye326 committed:
style: AuthController 불필요한 공백 라인 제거 및 코드 정리

jye326 committed:
refactor: FixtureUtil 및 AuthServiceTest 리팩토링

- FixtureUtil에 LocalDateTime 초기화 시 ZoneId 적용
- getVerifiedPhoneVerification 메서드 수정으로 인증 시간 변경 (3분 → 15분)
- AuthServiceTest에서 FixtureUtil 활용으로 코드 중복 제거 및 가독성 개선

jye326 committed:
refactor: Member 조회 및 예외 처리 개선

- NotFoundMemberException -> MemberNotFoundException 교체
- getMemberByLoginId 메서드 재사용으로 코드 중복 제거
- 관련 테스트 코드 수정 및 예외 검증 클래스 변경

jye326 committed:
refactor: AuthController URL 명칭 변경 및 테스트 코드 상태 수정

- find-login-id 엔드포인트를 login-id로 변경
- 테스트에서 잘못된 상태 코드 검증 (400 → 404) 수정 및 확인 로직 보완

jye326 committed:
Merge branch 'develop' into feature/1159-idpw-find

# Conflicts:
#	backend/fittoring/src/main/java/fittoring/application/reservation/presentation/ReservationController.java
#	backend/fittoring/src/test/java/fittoring/application/reservation/service/ReservationServiceTest.java

jye326 committed:
Merge remote-tracking branch 'origin/feature/1159-idpw-find' into feature/1159-idpw-find

jye326 committed:
Merge branch 'develop' into feature/1159-idpw-find

# Conflicts:
#	backend/fittoring/src/main/java/fittoring/application/auth/presentation/AuthController.java
#	backend/fittoring/src/main/java/fittoring/application/auth/service/AuthService.java
#	backend/fittoring/src/test/java/fittoring/integration/AuthIntegrationTest.java

dye0p reviewed (approved):
코멘트 반영이 어느정도 된 것 같으니 이만 승인하겠습니다!
